### PR TITLE
Brief housekeeping in the SDK

### PR DESCRIPTION
**Pull request by jmg-duarte**
- Rename the `HttpClient` to `ActyxClient`
- Remove the thread-based exponential backoff

### CODE REVIEW

File: rust/sdk/src/arb.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+// Used by trees::arb
+// Can't really be moved to `trees` since Rust disallows
+// implementing foreign traits on foreign types
```
Comment (rkuhn) [inline rust/sdk/src/arb.rs:3]: yeah, this is a peculiarity of the Rust way of doing things, which leads to good libraries implementing Arbitrary even though they don’t use it internally

File: rust/sdk/src/client.rs [hunk 2]
```diff
@@ -115,8 +115,11 @@ impl HttpClient {
         Ok(token)
     }
 
-    /// Makes request to Actyx apis. On http authorization error tries to
-    /// re-authenticate and retries the request ten times with increasing delay.
+    /// Perform a request (to Actyx APIs).
+    /// If an authorization error (code 401) is returned, it will try to re-authenticate.
+    /// If the service is unavailable (code 503), this method will retry to perform the
+    /// request up to 10 times with exponentially increasing delay - currently,
+    /// this behavior is only available if the `with-tokio` feature is enabled.
```
Comment (rkuhn) [inline rust/sdk/src/client.rs:122]: This text is good, but on a private method people won’t see it. I’d probably add it to each end-user method that calls this function.

**Review by rkuhn — APPROVED** [review]
LGTM apart from the doc comment

File: rust/sdk/src/client.rs (hunk 2 above)
Comment (jmg-duarte) [reply rust/sdk/src/client.rs:122]: I can add on them as well, this is a rust doc and not a regular one because we're the first clients after all

**Pull request merged**